spring-amqp/src/main/java/org/springframework/amqp/core/Queue.java

```diff
@@ -39,6 +39,9 @@
 
 	private final java.util.Map<java.lang.String, java.lang.Object> arguments;
 
+	private volatile String declaredName = AnonymousQueue.Base64UrlNamingStrategy.DEFAULT.generateName()
```

Comment:
Why can't we re-use an existing `name`?
If it is volatile `Queue` object, it doesn't matter what we have set to the property, but when it is an active, broker-connected entity, we will get a declared name there.

Anything specific in a favor of the separate property?
Thanks

Reply:
When the connection is lost and re-established, we can't declare the queue with a broker-declared name (illegal to name a queue starting with `amq.gen`. So we need to keep name as `""` so the broker will give us another new name.

Also, with the DMLC, I needed a unique value here in case the container is started before the queues could be declared - perhaps because the broker is down.